File: frontend/Services/DashboardService.cs
using System.Net.Http.Json;
using LendQ.Client.Models;

namespace LendQ.Client.Services;

public class DashboardService : IDashboardService
{
    private readonly HttpClient _http;

    public DashboardService(HttpClient http)
    {
        _http = http;
    }

    public async Task<DashboardSummary> GetSummaryAsync()
    {
        return await _http.GetFromJsonAsync<DashboardSummary>("dashboard/summary")
            ?? throw new InvalidOperationException("Dashboard summary was null.");
    }

    public async Task<List<LoanSummary>> GetDashboardLoansAsync()
    {
        return await _http.GetFromJsonAsync<List<LoanSummary>>("dashboard/loans")
            ?? new List<LoanSummary>();
    }

    public async Task<List<ActivityItem>> GetActivityAsync()
    {
        return await _http.GetFromJsonAsync<List<ActivityItem>>("dashboard/activity")
            ?? new List<ActivityItem>();
    }
}

File: frontend/src/app/core/services/dashboard.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable, catchError, of } from 'rxjs';
import { DashboardSummary, ActivityItem } from '../models/dashboard.model';
import { LoanSummary } from '../models/loan.model';

@Injectable({ providedIn: 'root' })
export class DashboardService {
  constructor(private http: HttpClient) {}

  getSummary(): Observable<DashboardSummary | null> {
    return this.http.get<DashboardSummary>('/api/v1/dashboard/summary').pipe(
      catchError(() => of(null))
    );
  }

  getLoans(): Observable<{ items: LoanSummary[] } | null> {
    return this.http.get<{ items: LoanSummary[] }>('/api/v1/dashboard/loans').pipe(
      catchError(() => of(null))
    );
  }

  getActivity(): Observable<{ items: ActivityItem[] } | null> {
    return this.http.get<{ items: ActivityItem[] }>('/api/v1/dashboard/activity').pipe(
      catchError(() => of(null))
    );
  }
}
